go/api/response_handlers.go

```diff
@@ -372,3 +373,24 @@ func handleStringSetResponse(response *C.struct_CommandResponse) (map[Result[str
 
 	return slice, nil
 }
+
+// Function to convert a map[string, V] to a value-key string array, used in Zadd
+func convertMapToKeyValueStringArray[V any](args map[string]V) []string {
+	result := make([]string, 0, len(args)*2)
+	for key, value := range args {
+		// Convert the value to a string after type checking
+		switch v := any(value).(type) {
+		case string:
+			result = append(result, v)
+		case int64:
+			result = append(result, strconv.FormatInt(v, 10))
+		case float64:
+			result = append(result, strconv.FormatFloat(v, 'f', -1, 64))
+		case bool:
+			result = append(result, strconv.FormatBool(v))
+		}
+		// Append the key
+		result = append(result, key)
+	}
+	return result
+}
```

Comment:
Zadd wants the pairs of values in the order of score-member pairs, and since scores can be duplicate values, we can only pass member-score maps as params which is why I added the function. I've chosen to move the function to under utils and kept it a separate function from the existing one.